src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -176,8 +180,20 @@ protected Bitmap doInBackground(Object... params) {
                 
                 if (mFile instanceof OCFile) {
                     thumbnail = doOCFileInBackground();
+
+                    if (((OCFile) mFile).isVideo()){
+                        thumbnail = addVideoOverlay(thumbnail);
+                    }
                 }  else if (mFile instanceof File) {
                     thumbnail = doFileInBackground();
+
+                    String url = ((File) mFile).getAbsolutePath();
+                    FileNameMap fileNameMap = URLConnection.getFileNameMap();
+                    String mMimeType = fileNameMap.getContentTypeFor("file://" + url);
```

Comment:
Please, use ```FileStorageUtils.getMimeTypeFromExtension(...)``` instead. That way we avoid introducing ```java.net.URLConnection``` in the imports. Let's prevent the temptation of using other network libraries.